src/server/game/Entities/Player/Player.cpp

```diff
@@ -25665,11 +25665,47 @@ void Player::StoreLootItem(uint8 lootSlot, Loot* loot, AELootResult* aeResult/*
         SendEquipError(msg, nullptr, nullptr, item->itemid);
 }
 
-bool Player::CanFlyInZone(uint32 mapid, uint32 zone) const
+bool Player::CanFlyInZone(uint32 mapid, uint32 areaid) const
 {
     // continent checked in SpellInfo::CheckLocation at cast and area update
-    uint32 v_map = sDB2Manager.GetVirtualMapForMapAndZone(mapid, zone);
-    return v_map != 571 || HasSpell(54197); // 54197 = Cold Weather Flying
+    bool can_fly = false;
+    switch (mapid)
```

Comment:
where is v_map now ?

Reply:
GetVirtualMapForMapAndZone returns strange values, I don't think it's meant for flying, rather for in-game map.
WorldMapArea.db2 - field18 (DisplayMapID)